src/services/factories/tokenList.ts

```diff
@@ -32,7 +42,7 @@ export function getTokensFactory(factoryParams: {
     }
   }
 
-  async function updateWithRetries(
+  async function updateTokenIds(
     networkId: number,
     tokens: TokenDetails[],
     maxRetries = 3,
```

Comment:
why it needs retry logic here? it was refactored to an util, right?

Reply:
1. Historical reasons. On the previous version, it was the first network call that could cause failures. With this change, there is at least one other call before that can fail/retried
2. It all started from a TODO to only retry fetching the ids for tokens that failed, instead of query all tokens again. So, this is a sort of a custom retry function, doesn't quite fit into the generic one.